api/gateway/basecommerce/basecommerce.cfc

```diff
@@ -0,0 +1,134 @@
+component
+	displayname='BaseCommerce Interface'
+	output=false
+	extends='cfpayment.api.gateway.base' {
+
+	variables.cfpayment.GATEWAY_NAME = 'BaseCommerce';
+	variables.cfpayment.GATEWAY_VERSION = '1.0';
+	
+	public string function getProcessorID() {
+		return variables.cfpayment.ProcessorID;
+	}
+
+	//Implement primary methods
+	public any function purchase(required any money, struct options=structNew()) {
+		arguments.options.batType = 'XS_BAT_TYPE_DEBIT';
+		return populateResponse(transactionData(argumentcollection = arguments));
+	}
+	
+	public any function credit(required any money, struct options=structNew()) {
+		arguments.options.batType = 'XS_BAT_TYPE_CREDIT';
+		return populateResponse(transactionData(argumentcollection = arguments));
+	}
+	
+	public any function store(required any account, struct options=structNew()) {
+		return populateResponse(accountData(account, options));
+	}
+
+	//Private Functions
+	private any function populateResponse(required struct data) {
+		local.response = new cfpayment.api.model.response(service = getService());
```

Comment:
the base gateway has an existing createResponse() method that would probably be better to use here rather than create a (potentially) fragile link to a specific cfc.